### DISCUSSION TITLE
Why is it so heavy?

### DISCUSSION THREAD

**Discussion by lildadou** [Koenkk/zigbee2mqtt#21211] [General]
My question is not a reproach but pure intellectual curiosity. I'm a software engineer myself, and I know that there could be any number of reasons for the decisions we make. Here, I can't think of any, and I'd like to satisfy my curiosity ^^.

When I discovered this project, I thought "great! I'll be able to put my usb dongle on my wifi router, which is right in the middle of my house, and relay its events, wired, to the server in my basement". But I only had 10 MiB on my wifi router and the Docker image, although based on Alpine, already weighed over 30 MiB... I then realized that I'd made an error of judgement.

The story has a happy ending, as I was finally able to do USB-over-IP with less than one MiB, but I still want to understand why I made this error of judgement. What is zigbee2mqtt designed for and why is it so heavy?

**Comment by Koenkk** [comment]
This footprint mainly comes from nodejs

**Comment by lildadou** [comment]
Yes, I saw that so the question becomes "why nodejs?" But that's not really what I'm interested in; what I wanted to know was the usage scenarios imagined. I'm just starting out in home automation and I'm a bit stuck on my "remote antenna" idea; all this is hindering my understanding of the need the software was intended to meet.

**Reply by Koenkk** [reply]
It is meant to be run on a Raspberry Pi/NUC, not on a router which such limited resources.
